### DISCUSSION TITLE
docs/schema-generator

### DISCUSSION THREAD

**Discussion by giscus** [mikro-orm/mikro-orm#6298] [Comments]
# docs/schema-generator

SchemaGenerator can do harm to your database. It will drop or alter tables, indexes, sequences and such. Please use this tool with caution in development and not on a production server. It is meant for helping you develop your Database Schema, but NOT with migrating schema from A to B in production. A safe approach would be generating the SQL on development server and saving it into SQL Migration files that are executed manually on the production server.

https://mikro-orm.io/docs/schema-generator

**Comment by amerkarim** [comment]
If I need to generate the update sql script from specific migrations only, how can I do that?
For example, if I have 10 migrations called migration1 to migration10, I want to generate the sql script of say migration7 to the latest migration10, how to do that?

In EF core, there's a command that looks like this: "dotnet ef migrations script fromMigration toMigration", the toMigration is optional and the default is the latest one.

**Reply by B4nan** [reply]
If you mean running specific migrations, you can do that via `--from` and `--to` parameters.

https://mikro-orm.io/docs/migrations#using-via-cli

**Reply by amerkarim** [reply]
No not running specific ones, just grabbing the SQL script for the --from and --to migrations.
